src/api/java/jni/solver.cpp

```diff
@@ -978,6 +979,25 @@ Java_io_github_cvc5_Solver_declareOracleFun(JNIEnv* env,
   CVC5_JAVA_API_TRY_CATCH_END_RETURN(env, 0);
 }
 
+/*
+ * Class:     io_github_cvc5_Solver
+ * Method:    addPlugin
+ * Signature: (JLio/github/cvc5/AbstractPlugin;)V
+ */
+JNIEXPORT void JNICALL Java_io_github_cvc5_Solver_addPlugin(JNIEnv* env,
+                                                            jobject,
+                                                            jlong pointer,
+                                                            jobject plugin)
+{
+  CVC5_JAVA_API_TRY_CATCH_BEGIN;
+  Solver* solver = reinterpret_cast<Solver*>(pointer);
+  jobject pluginReference = env->NewGlobalRef(plugin);
+  globalReferences[pointer].push_back(pluginReference);
+  ApiPlugin* p = new ApiPlugin(solver->getTermManager(), env, pluginReference);
```

Comment:
I have two questions in this line:
- Is it ok, to use term manager associated with the solver? I am not sure why Plugin needs a term manager in its constructor @ajreynol .
- @aniemetz  we have to use a pointer here for the plugin and there is a memory leak here. Any suggestion for deleting this pointer other than global variable.